app/core/wallet.js

```diff
@@ -1,12 +1,25 @@
 // @flow
 import { wallet } from 'neon-js'
 import { map, extend } from 'lodash'
+import fetch from 'node-fetch'
 
 import { ASSETS } from './constants'
 import { toBigNumber } from './math'
 
 const MIN_PASSPHRASE_LEN = 4
 
+let addressBlacklist: Array<string> | null = null
+export const isInBlacklist = (address: string): Promise<boolean> => {
+  if (addressBlacklist !== null) return Promise.resolve(addressBlacklist.includes(address))
+  return fetch('https://raw.githubusercontent.com/CityOfZion/phishing/master/blockedAddresses.json')
```

Comment:
We could write this with axios and async/await syntax in a pretty direct manner:

```js
import axios from 'axios'

export const isInBlacklist = async (address: string): Promise<boolean> => {
  if (addressBlacklist === null) {
    const { data } = await axios.get('https://raw.githubusercontent.com/CityOfZion/phishing/master/blockedAddresses.json')
    addressBlacklist = data
  }
  return addressBlacklist.includes(address)
}
```